framework/storm-tracking-cryptographic-materials-cache.md

```diff
@@ -53,24 +63,41 @@ attempts will be made to refresh the cache.
 
 This should be significantly less than the TTL for any item put into the cache.
 
+The Grace Period should be at least a few times larger than the Grace Interval,
+so that if something goes wrong with the first request, a second or third request will be made.
+
 ### Grace Interval
 
 A number of seconds (at least 1, default 1).
 
-While within the [grace period](#grace-period),
-attempts to refresh the cache are made no more often than once per interval.
+An entry that has been in flight for this long is no longer considered in flight.
+
+While within the [grace period](#grace-period), after this period another request is made.
+
+While doing the initial fetch for a key, after this period another request is made.
+
+The Grace Interval should be larger than the longest time expected for a request to the underlying cryptographic materials provider.
 
 ### FanOut
 
 A number (at least 1, default 20).
 
 The maximum number of individual keys for which lookups can be in flight.
 
+Other requests are blocked until the number of outstanding requests drops below this number,
```

Comment:
I think you should clarify what we mean by requests here. Because if the request is for exactly the same ID, then it is not blocked. it would be returned. This has to do with requests to different ids yes?